python/ray/autoscaler/gcp/config.py

```diff
@@ -104,16 +103,65 @@ def generate_rsa_key_pair():
     return public_key, pem
 
 
+def _create_crm(config, gcp_credentials):
+    return discovery.build(
+        "cloudresourcemanager", "v1", credentials=gcp_credentials)
+
+
+def _create_iam(config, gcp_credentials):
+    return discovery.build("iam", "v1", credentials=gcp_credentials)
+
+
+def _create_compute(config, gcp_credentials):
+    return discovery.build("compute", "v1", credentials=gcp_credentials)
+
+
+def fetch_gcp_credentials_from_provider_config(provider_config):
+    """
+    Attempt to fetch and parse the JSON GCP credentials from the provider
+    config yaml file.
+    """
+    service_account_info_string = provider_config.get("gcp_credentials")
+    if service_account_info_string is None:
+        logger.info("gcp_credentials not found in cluster yaml file. "
+                    "Falling back to GOOGLE_APPLICATION_CREDENTIALS "
+                    "environment variable.")
+        # If gcp_credentials is None, then discovery.build will search for
+        # credentials in the local environment.
+        return None
+    else:
+        # If parsing the gcp_credentials failed, then the user likely made a
+        # mistake in copying the credentials into the config yaml.
+        try:
+            service_account_info = json.loads(service_account_info_string)
+        except json.decoder.JSONDecodeError:
+            logger.warning(
```

Comment:
Should this just be a blocking error? It seems like if a user tried to define this and it didn't work we should bail at config parse time, not sometime later with an unrelated google authentication error.